test/e2e/manifestworkreplicaset_test.go

```diff
@@ -303,46 +303,55 @@ var _ = ginkgo.Describe("Test ManifestWorkReplicaSet", func() {
 			gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
 			ginkgo.By("Check manifestWork replicaSet status is updated")
-			gomega.Eventually(func() bool {
+			gomega.Eventually(func() error {
 				mwrSet, err := t.HubWorkClient.WorkV1alpha1().ManifestWorkReplicaSets(namespace).Get(context.TODO(), mwReplicaSetName, metav1.GetOptions{})
 				if err != nil {
-					return false
+					return err
 				}
 
 				if !meta.IsStatusConditionTrue(mwrSet.Status.Conditions, workapiv1alpha1.ManifestWorkReplicaSetConditionPlacementVerified) {
-					return false
+					return err
 				}
 
-				return mwrSet.Status.Summary.Total == numOfClusters
-			}, t.EventuallyTimeout*5, t.EventuallyInterval*5).Should(gomega.BeTrue())
+				if mwReplicaSet.Status.Summary.Total != numOfClusters {
+					return fmt.Errorf("total number of clusters is not correct, expect %d, got %d", numOfClusters, mwReplicaSet.Status.Summary.Total)
+				}
+				return nil
+			}, t.EventuallyTimeout*10, t.EventuallyInterval*10).Should(gomega.Succeed())
```

Comment:
this seems really long...why do we need 10 times?

Reply:
Yes, but think in another way, we can move the **Enable** and **Disable** the **ReplicaSet** at the beginning to save some time.